### PR TITLE
[BACKLOG-1561] - Add the ability to get security/ACL information about a datasource

### PR DESCRIPTION
**Pull request by aliakseihaidukou**
adopt the code to new pentaho-platform approach, clean cache for metadata
fix metadata service - return 401 status if user doesn't have permissions and ds does not exist, unit-test for this
fix unit-tests

@pentaho-nbaker please review

### CODE REVIEW

**Comment by aliakseihaidukou** [comment]
switched permissions from administer security to datasource management for get/set ACL, tests
fixed license year, added license text if missed

**Comment by pentaho-nbaker** [comment]
Please start squashing commits together. I'll review this today.

File: src/org/pentaho/platform/dataaccess/datasource/api/DataSourceWizardService.java [hunk 1]
```diff
@@ -69,6 +70,9 @@
   protected IModelerService modelerService;
   protected IDatasourceMgmtService datasourceMgmtSvc;
 
+  protected IAclAwarePentahoMetadataDomainRepositoryImporter aclMetadataHelper;
```
Comment (pentaho-nbaker) [inline src/org/pentaho/platform/dataaccess/datasource/api/DataSourceWizardService.java]: The "helper" portion of these names is confusing. Nothing huge, just clean it up next time you're in there please

File: src/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java [hunk 2]
```diff
@@ -25,41 +25,47 @@
...
+  private IMetadataDomainRepository metadataDomainRepositoryReal;
   protected IMondrianCatalogService mondrianCatalogService;
-  protected FileService fileService;
-  protected IAclNodeHelper aclHelper;
   protected RepositoryFileAclAdapter repositoryFileAclAdapter;
+  private MondrianCatalogRepositoryHelper mondrianCatalogRepositoryHelper;
 
   public DatasourceService() {
-    metadataDomainRepository = PentahoSystem.get( IMetadataDomainRepository.class, PentahoSessionHolder.getSession() );
+    metadataDomainRepositoryReal = PentahoSystem.get( IMetadataDomainRepository.class, PentahoSessionHolder.getSession() );
```
Comment (pentaho-nbaker) [inline src/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java]: Please add comments explaining why this is done.

File: src/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java [hunk 3]
```diff
@@ -43,9 +46,16 @@
 import com.sun.jersey.multipart.FormDataBodyPart;
 
 public class MetadataService extends DatasourceService {
+  protected IAclAwarePentahoMetadataDomainRepositoryImporter aclMetadataHelper;
```
Comment (pentaho-nbaker) [inline src/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java]: Naming confusion again.

File: src/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java [hunk 4]
```diff
@@ -105,21 +115,39 @@ public void importMetadataDatasource( String domainId, InputStream metadataFile,
...
 
-  public RepositoryFileAclDto getMetadataAcl( String domainId ) throws PentahoAccessControlException {
-    if ( !canAdministerCheck() ) {
-      throw new PentahoAccessControlException();
+  public RepositoryFileAclDto getMetadataAcl( String domainId )
+      throws PentahoAccessControlException, FileNotFoundException {
+    checkMetadataExists( domainId );
+    if ( aclMetadataHelper != null ) {
+      final RepositoryFile domainRepositoryFile = aclMetadataHelper.getDomainRepositoryFile( domainId );
+      final RepositoryFileAcl acl = aclMetadataHelper.getAclHelper().getAclFor( domainRepositoryFile );
```
Comment (pentaho-nbaker) [inline src/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java]: A couple things. 

I don't like getAclHelper(). These methods should be on the ACL Aware interfaces directly.

getAclFor( RepositoryFile):
Move this to the ACLAware interface and change it to take the domainId String if possible. I don't like fetching the RepositoryFile and passing it back in, the interface should use the logical DomainID wherever possible

File: src/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java (hunk 2 above)
Comment (aliakseihaidukou) [inline src/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java]: Nick, comments for this are near declaring of these fields - lines 43, 45. Do I need to add more info?

Comment (pentaho-nbaker) [inline src/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java]: We very rarely, almost never, directly construct an Implementation like this. What happens if an OEM has their own MetadataDomainRepository which needs to be used? A better solution is query for the "real" and the "plain" beans

metadataDomainRepositoryReal  = PentahoSystem.get( IMetadataDomainRepository.class, "IMetadataDomainRepositoryImpl", PentahoSessionHolder.getSession() );

metadataDomainRepository = PentahoSystem.get( IMetadataDomainRepository.class, "IMetadataDomainRepository", PentahoSessionHolder.getSession() );

Comment (aliakseihaidukou) [inline src/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java]: I saw such usage of PentahoMetadataDomainRepository in MetadataDatasourceService. When I will move get/set ACL methods to ACL Aware interface, I will remove this metadataDomainRepository/metadataDomainRepositoryReal couple leaving only metadataDomainRepository = PentahoSystem.get( IMetadataDomainRepository.class, PentahoSessionHolder.getSession() );

**Pull request merged**